[lldb][Target] Refactor Target::FindLoadAddr

chelcassanova opened the pull request:
`Target::FindLoadAddrForNameInSymbolsAndPersistentVariables` has been split into `Target::FindLoadAddrForNameInSymbols` and `Target::FindLoadAddrForNameInPersistentVariables`. Instead of performing both searches in one function,
`Target::FindLoadAddrForNameInSymbolsAndPersistentVariables` will call its constituent functions.

This allows `SwiftLanguageRuntime::FindSymbolForSwiftObject` to look for symbols with `Target::FindLoadAddrForNameInSymbols` instead of performing this search on its own.

rdar://32228544

chelcassanova committed:
[lldb][Target] Refactor Target::FindLoadAddr

`Target::FindLoadAddrForNameInSymbolsAndPersistentVariables` has been
split into `Target::FindLoadAddrForNameInSymbols` and
`Target::FindLoadAddrForNameInPersistentVariables`. Instead of
performing both searches in one function,
`Target::FindLoadAddrForNameInSymbolsAndPersistentVariables` will call
its constituent functions.

This allows `SwiftLanguageRuntime::FindSymbolForSwiftObject` to look for symbols
with `Target::FindLoadAddrForNameInSymbols` instead of performing this
search on its own.

rdar://32228544

JDevlieghere commented:
This looks like generic code, should this target llvm.org rather than the Swift fork?

chelcassanova commented:
`FindLoadAddrForNameInSymbolsAndPersistentVariables` has no usage in the codebase and can removed, the same goes for `FindLoadAddrForNameInPersistentVariables`. `SwiftLanguageRuntime::FindSymbolForSwiftObject` is using `FindLoadAddrForNameInSymbols` but if that's the only place that `FindLoadAddrForNameInSymbols` would get used then we could go back to the original implementation of `FindSymbolForSwiftObject` to remove `FindLoadAddrForNameInSymbolsAndPersistentVariables` altogether.

chelcassanova committed:
Remove `Target::FindLoadAddrForNameInSymbolsAndPersistentVariables`

This reverts commit 2542856f5b7c61f2389a0cbe6ade25f5b5ac8ae6. Remove
`Target::FindLoadAddrForNameInSymbolsAndPersistentVariables` as this is
going unused in the codebase. As such the implementation being used in
`FindSymbolForSwiftObject` is reverted to its original version.

JDevlieghere commented:
@swift-ci test
